lib/streamlit/elements/image.py

```diff
@@ -386,6 +387,42 @@ def image_to_url(
         return ""
 
 
+def _get_svg_width_and_height(image: str) -> Tuple[str, str]:
+    root_idx = image.find(">")
+    root = image[:root_idx]
+    width = "100"
+    height = "100"
+    vb = root.find("viewBox")
+    if vb != -1:
+        view_box = [
+            val.replace('"', "") for val in root[vb + len("viewBox=") + 1 :].split(" ")
+        ][:4]
+        width, height = view_box[2], view_box[3]
+    return width, height
+
+
+def _add_svg_attr(image: str, attr_name: str, attr_value: str) -> str:
+    root_idx = image.find(">")
+    root = image[:root_idx]
+    if root.find(attr_name) == -1:
+        width = f'{attr_name}="{attr_value}"'
+        image = f"{root} {width}{image[root_idx:]}"
+    return image
+
+
+def _normalize_svg(image: str) -> str:
+    width, height = _get_svg_width_and_height(image)
+    root_idx = image.find(">")
+    root = image[:root_idx]
+    if root.find("width") == -1:
+        image = _add_svg_attr(image, "width", width)
+    if root.find("height") == -1:
+        image = _add_svg_attr(image, "height", height)
+    if root.find("xmlns") == -1:
+        image = _add_svg_attr(image, "xmlns", "http://www.w3.org/2000/svg")
```

Comment:
Is this something required in the SVG? Is the `img` tag not able to render the SVG if this is missing?

Reply:
When `xmlns` attribute is missing inside SVG the SVG is not rendered when is passed as an url into `img` HTML tag, however when it is rendered as markup `xmlns` attribute is not required. 

When it comes to `height` and `width` attributes, to make **streamlit** scale SVGs I've noticed that since on `st_image` we've only have optional width attribute there's no need to setup `height` on SVG. Streamlit scaling needs only width.

Thus I've removed setting the `height` attribute from this PR 👍